library/common/http/dispatcher.h

```diff
@@ -32,10 +32,49 @@ class Dispatcher : public Logger::Loggable<Logger::Id::http> {
    * @return envoy_stream_t handle to the stream being created.
    */
   envoy_status_t startStream(envoy_stream_t stream, envoy_observer observer);
+
+  /**
+   * Send headers over an open HTTP stream. This method can be invoked once and needs to be called
+   * before send_data.
+   * @param stream, the stream to send headers over.
+   * @param headers, the headers to send.
+   * @param end_stream, indicates whether to close the stream locally after sending this frame.
+   * @return envoy_status_t, the resulting status of the operation.
+   */
   envoy_status_t sendHeaders(envoy_stream_t stream, envoy_headers headers, bool end_stream);
+
+  /**
+   * Send data over an open HTTP stream. This method can be invoked multiple times.
+   * @param stream, the stream to send data over.
+   * @param data, the data to send.
+   * @param end_stream, indicates whether to close the stream locally after sending this frame.
+   * @return envoy_status_t, the resulting status of the operation.
+   */
   envoy_status_t sendData(envoy_stream_t stream, envoy_data data, bool end_stream);
-  envoy_status_t sendMetadata(envoy_stream_t stream, envoy_headers headers, bool end_stream);
-  envoy_status_t sendTrailers(envoy_stream_t stream, envoy_headers headers);
+
+  /**
+   * Send metadata over an HTTP stream. This method can be invoked multiple times.
+   * @param stream, the stream to send metadata over.
+   * @param metadata, the metadata to send.
+   * @return envoy_status_t, the resulting status of the operation.
+   */
+  envoy_status_t sendMetadata(envoy_stream_t stream, envoy_headers metadata);
```

Comment:
Does the impl also need to be updated? (Or _should_ it be updated?)

Reply:
not implemented yet, so the cc file has no argument names, only types.